neuraxle/base.py

```diff
@@ -1494,8 +1494,11 @@ class NonFittableMixin:
     Note: fit methods are not implemented
     """
 
-    def handle_fit_transform(self, data_container: DataContainer, context: ExecutionContext):
-        return self, self.handle_transform(data_container, context)
+    def handle_fit(self, data_container: DataContainer, context: ExecutionContext) -> ('BaseStep', DataContainer):
+        return self, data_container
+
+    def handle_fit_transform(self, data_container: DataContainer, context: ExecutionContext) -> ('BaseStep', DataContainer):
+        return self, data_container
```

Comment:
The hyperparameters could potentially change things in the transform event if it is non-fittable. I think it needs to re-hash with hyperparams.

Reply:
I would forbid hyperparameters in these wrappers... users should not be setting hyperparams for these wrapper because they only change the execution flow

Reply:
maybe just overriding init to make sure no one can set hyperparams for these

Reply:
That is not true. 

For example our MultiplyByN step is a non-fittable mixin, but it changes the values and hence everything **must** be re-hashed thereafter. Otherwise the pipeline will bug and reuse bad cache with a different multiplication. 

Please do the change, except if I'm saying something very wrong I think it works like I say.

Reply:
It is totally normal to have nonfittablemixins that has hyperparameters and that transform data, even if it doesn't fit.

Reply:
Would deleting those overrided methods and letting BaseStep handle everything with the `handle_*` methods work? It'd need the default hasher.

Reply:
@guillaume-chevalier this is not a mixin... it is a wrapper for an execution mode. that wouldn't make sense that someone adds any behavior to this wrapper.